torchgeo/datasets/mmflood.py

```diff
@@ -0,0 +1,458 @@
+
+        Returns:
+            dict containing list of paths, with 'image', 'dem' and 'mask' as keys
+        """
+        paths = {}
+        dirpath = os.path.join(self.root, self.metadata['directory'])
+        # initialize tif file lists containing masks, DEM and S1_raw data
+        if load_all:
+            # Get all directories
+            folders = os.listdir(dirpath)
+        else:
+            # Assemble regex for glob
+            folders = (
+                self.metadata_df[self.metadata_df['subset'] == self.split].index + '-*'
+            ).tolist()
+
+        image_files = []
+        mask_files = []
+        dem_files = []
+        for f in folders:
+            path = os.path.join(self.root, self.metadata['directory'], f)
+            image_files += glob(os.path.join(path, 's1_raw', '*.tif'))
+            mask_files += glob(os.path.join(path, 'mask', '*.tif'))
+            dem_files += glob(os.path.join(path, 'DEM', '*.tif'))
+
+        image_files = sorted(image_files)
+        mask_files = sorted(mask_files)
+        dem_files = sorted(dem_files)
+
+        paths['image'] = image_files
+        paths['mask'] = mask_files
+        paths['dem'] = dem_files
+
+        # Verify image, dem and mask lengths
+        assert (
+            len(paths['image']) > 0
+        ), f'No images found, is the given path correct? ({self.root!s})'
+        assert (
+            len(paths['image']) == len(paths['mask'])
+        ), f'Length mismatch between tiles and masks: {len(paths["image"])} != {len(paths["mask"])}'
+        assert len(paths['image']) == len(
+            paths['dem']
+        ), 'Length mismatch between tiles and DEMs'
+
+        if check_folders:
+            # Verify image, dem and mask pairings
+            self._verify_pairings(paths['image'], paths['dem'], paths['mask'])
+
+        return paths
+
+    def _load_folders(self) -> list[dict[str, str]]:
+        """Load folder paths.
+
+        Returns:
+            list of dicts of s1, dem and masks folder paths
+        """
+        paths = self._load_tif_files(check_folders=False, load_all=False)
+
+        res_folders = [
+            {'s1_raw': img_path, 'mask': mask_path, 'dem': dem_path}
+            for img_path, mask_path, dem_path in zip(
+                paths['image'], paths['mask'], paths['dem']
+            )
+        ]
+
+        return res_folders
+
+    def _load_image(self, index: list[int], query: BoundingBox) -> Tensor:
+        """Load a either a single image or a set of images, merging them.
+
+        Args:
+            index: indexes to return
+            query: (minx, maxx, miny, maxy, mint, maxt) coordinates to index
+
+        Returns:
+            the merged image
+        """
+        image = self._load_tif(index, modality='s1_raw', query=query).float()
+        if self.include_dem:
```

Comment:
In the docstring you mentioned that the DEM is not available for all regions, but is this also explicitly handled? Or what happens if the DEM is not available. Do I get a tensor with a zero padded channel dimension? 

Or maybe as an additional check, if you have the full dataset downloaded, can you instantiate a datamodule or dataloader over the dataset an iterate through the length of the dataset without errors?

Reply:
DEM is available for all the regions. Hydrography maps are missing for many of the tiles (738 missing). They are downloaded but not handled by this class. Maybe should I remove them from the docstring? Or introduce a flag `include_hydro` in the constructor and load only a portion of the dataset (1012 tiles out of 1748 total)?

Reply:
Ah sorry, misread. If you think that hydrography should be useful to users, then I think a flag to `include_hydro` with default `False` would be a good thing. And then just add in the docstring like you mentioned here, that if someone wants to use hydro they will only have a subset of the data available.